nntrainer/src/neuralnet.cpp

```diff
@@ -581,6 +579,40 @@ void NeuralNetwork::readModel() {
   ml_logi("read modelfile: %s", model.c_str());
 }
 
+/**
+ * @brief     Run NeuralNetwork train
+ */
+int NeuralNetwork::train(
+  bool (*train_func)(std::vector<std::vector<std::vector<float>>> &,
+                     std::vector<std::vector<std::vector<float>>> &, int &),
```

Comment:
How about using typedef? For example, in header,
```
typedef bool (*BufferFunc)(std::vector<std::vector<std::vector<float>>> &, std::vector<std::vector<std::vector<float>>> &, int &);
```
Then, you may write this
```
int NeuralNetwork::train (BufferFunc train_func, BufferFunc val_func, BufferFunc test_func);
```

Reply:
Thanks for the tip. But I prefer to use typedef  std::vector<std::vector<std::vector<float>>> vec_3d. Cause it makes more concise the other place too.